### ISSUE TITLE
Support Complex Numbers?

### ISSUE THREAD

**Issue by AndiMD** [SciML/DiffEqOperators.jl#237]
Can we allow complex numbers? Example:
```
RobinBC( (0.5+0im,0.5*im, 1.0+0im), (0.5+0im,0.5*im, 0.0+0im), 1.0)

ERROR: LoadError: MethodError: no method matching RobinBC(::Tuple{Complex{Float64},Complex{Float64},Complex{Float64}}, ::Tuple{Complex{Float64},Complex{Float64},Complex{Float64}}, ::Float64)
```

**Comment by ChrisRackauckas** [comment]
Yes, we should probably just change the type signatures to allow this (and test it of course)

**Comment by AndiMD** [comment]
this will also allow optimizing boundaries with `ForwardDiff.Dual`, which appears to be used by certain ODE solvers.

**Comment by ChrisRackauckas** [comment]
Yes, I would not be opposed to seeing this change.

**Comment by ChrisRackauckas** [comment]
It should probably just take changing the type signatures all to `<:Number`

**Comment by volkerkarle** [comment]
Hi folks, may I ask what is the progress for this and if there is something to do? I wonder what is the technical current state, can we implement complex PDEs such as Gross–Pitaevskii equations?
Thanks!
